driver-core/src/main/com/mongodb/async/AsyncBatchCursor.java

```diff
@@ -41,6 +41,18 @@
     void next(SingleResultCallback<List<T>> callback);
 
     /**
+     * A special {@code next()} case that returns the next batch if available or null.
+     *
+     * <p>Tailable cursors are an example where this is useful. A call to {@code tryNext()} may return null, but in the future calling
+     * {@code tryNext()} would return a new batch if a document had been added to the capped collection.</p>
+     *
+     * @param callback callback to receive the next batch of results
+     * @since 3.6
+     * @mongodb.driver.manual reference/glossary/#term-tailable-cursor Tailable Cursor
+     */
+    void tryNext(SingleResultCallback<List<T>> callback);
```

Comment:
It's a good idea for the callback-based API.  I imagine it's not useful in reactive stream style usage?